### PR TITLE
Enhancement to support multi-platform to mutate a file

### PR DESCRIPTION
**Pull request by alexchapar**
Modifications on Stryker.ts
- Add a function to get the relative path of the file to mutate based from the workspace folder.
- Use the FileSystem Path ("fsPath"), instead of "path", to mutate a file to support multi-platform
- Update the unit tests
- Was tested on windows (and on Ubuntu)

Update .husky scripts to support Windows

It is associated to the issue #183

### CODE REVIEW

**Comment by alexchapar** [comment]
I had an issue with husky but was too late to fix it 😬 and one test was missing along the way
So I added the missing test, fix/improve husky to support windows and rebase to the main branch... now the gating is good 😁

File: src/stryker.ts [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
 import { makeReusableTerminal, runCommand } from './terminal';
+import path from 'path';
 
 export type CommandRunner = (args: { file: vscode.Uri; lineRange?: string }) => void;
 
+const getRelativePath = (file: vscode.Uri): string => {
+  const workspaceFolder = vscode.workspace.getWorkspaceFolder(file);
+  if (workspaceFolder) {
+    return path.relative(workspaceFolder.uri.fsPath, file.fsPath);
+  } else {
```
Comment (slcp) [inline src/stryker.ts:12]: Suggestion: we could lose this `else` block entirely as there is an early return in the initial `if` block.

```
if (workspaceFolder) {
    return path.relative(workspaceFolder.uri.fsPath, file.fsPath);
  }
return file.fsPath;
```

**Review by slcp — COMMENTED** [review]
One minor suggestion. Otherwise, I'm struggling to get the test suite to pass locally which I'd like to resolve before moving this forward but by the looks of it this PR achieves what is required for the compatibility issues you described on Windows. I'll. come back to this when I understand why the tests are failing on my end.

**Comment by alexchapar** [comment]
> One minor suggestion. Otherwise, I'm struggling to get the test suite to pass locally which I'd like to resolve before moving this forward but by the looks of it this PR achieves what is required for the compatibility issues you described on Windows. I'll. come back to this when I understand why the tests are failing on my end.

Based on your struggles, I did experience some issues too (and I thought it was a Windows problem...) ... What I did was clean the root (remove node_modules, dist, out, strykerTmp)... Reinstall all using yarn install... Also I remove all the explicit "yarn" in the package.json scripts (eg.:  "test:unit": "jest \"(?<=.unit).test.ts$\" --coverage")... Then I was able to make it work on my machine... I'm unsure, but I think it's a mix of the strykerTmp (created and used with the --incremental) that seems to create the issue... Then again, I'm not an expert on this 😅 ...

**Comment by alexchapar** [comment]
> One minor suggestion. Otherwise, I'm struggling to get the test suite to pass locally which I'd like to resolve before moving this forward but by the looks of it this PR achieves what is required for the compatibility issues you described on Windows. I'll. come back to this when I understand why the tests are failing on my end.

Oh! and another thing I remember now, which I forgot to write earlier... When I had problems testing the extension at runtime (in vscode's extension development host), I had the extension installed from the marketplace... So I uninstalled the extension to make sure it didn't cause false errors when launched.

File: src/stryker.ts (hunk 1 above)
Comment (alexchapar) [reply src/stryker.ts]: I applied your suggestion ☺

**Review by slcp — APPROVED** [review]
Thank you for this work :)

**Comment by slcp** [comment]
Merging this. I've had a peek at the work you are doing on an extension for .Net Stryker that alludes to possible support for other ecosystems. It would be interesting to hear any thoughts you might have of supporting JS/TS as this extension does :) Create a new issue to discuss them.

**Comment by alexchapar** [comment]
> Merging this. I've had a peek at the work you are doing on an extension for .Net Stryker that alludes to possible support for other ecosystems. It would be interesting to hear any thoughts you might have of supporting JS/TS as this extension does :) Create a new issue to discuss them.

Thanks, and sorry for the delay (little rushes occurred) and yes, it could be interesting to have something in common... I'm still figuring out how to code some parts in TS (🤔 still learning 😋), to make the .Net experience "enjoyable"

**Pull request merged**